### PR TITLE
feat(microservices): added MqttOption for passing raw output to pub

### PR DESCRIPTION
**Pull request by notusertelken**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
mqtt-server and client stringify the output of the serializer before emitting/publishing events

Issue Number: [7288](https://github.com/nestjs/nest/issues/7288)

## What is the new behavior?
moved JSON.stringify to the serializer, so now the output of the serializer goes directly into publish/emit.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: packages/microservices/interfaces/microservice-configuration.interface.ts [hunk 1]
```diff
@@ -119,6 +119,7 @@ export interface MqttOptions {
       rh?: number;
     };
     userProperties?: Record<string, string | string[]>;
+    rawOutputPackets?: boolean;
```
Comment (kamilmysliwiec) [inline packages/microservices/interfaces/microservice-configuration.interface.ts:122]: As this is a custom configuration property that we want to add on top of the MQTT-specific settings, can we also add a comment here explaining what's the use-case/purpose of this flag?

Comment (notusertelken) [reply packages/microservices/interfaces/microservice-configuration.interface.ts:122]: will do refactor to move the stringify to the serializers so the options remain MQTT-specific

**Comment by notusertelken** [comment]
If this implementation is found acceptable, ill shortly make PRs for the other transport servers

**Comment by kamilmysliwiec** [comment]
@notusertelken looks good!

**Comment by notusertelken** [comment]
had forgotten about the client :sweat_smile:

**Comment by kamilmysliwiec** [comment]
LGTM

**Comment by tajnymag** [comment]
The server/message parsing isn't that much of a problem, as the binary payload get be retrieved with

```ts
context.getPacket().payload as Buffer
```

**Review by benjGam — APPROVED** [review]

**Pull request merged**